lntypes/preimage.go

```diff
@@ -0,0 +1,59 @@
+package lntypes
```

Comment:
Not the biggest fan of a package like this...but I guess it works

Reply:
Yes, if it annoys us after some time, we can do something about it. Most important is that we have the strict typing now on hash and preimage.